test/test_linalg.py

```diff
@@ -1353,7 +1353,7 @@ def test_inverse_errors(self, device, dtype):
         def run_test_singular_input(batch_dim, n):
             x = torch.eye(3, 3, dtype=dtype, device=device).reshape((1, 3, 3)).repeat(batch_dim, 1, 1)
             x[n, -1, -1] = 0
-            with self.assertRaisesRegex(RuntimeError, rf'For batch {n}: U\(3,3\) is zero'):
+            with self.assertRaisesRegex(RuntimeError, r'.*For batch .* : U(3,3) is zero.*'):
```

Comment:
Question: why exactly is the batch number not the same?

Also, I see that you've added a space between the `.*` and the `:`, and removed the backslashes from the parentheses; does the test pass with those changes? At first glance, I wouldn't expect it to.

In any case, I might rewrite the regex like this:
```suggestion
            with self.assertRaisesRegex(RuntimeError, r'.*For batch \d+: U\(3,3\) is zero.*\.'):
```

Reply:
The reason is if you look at the test error, the \( stays a \( instead of resolving into a (, which I think is because it is an rstring. This might be bad though, because then maybe the .* won't get picked up either :/ I'll look into this.

Reply:
Yea I am going to make it not an rstring. Because of the macos failure below.